File: ClothingWarehouseAPI/src/main/java/com/example/clothingwarehouseproject/domain/Inventory.java
package com.example.clothingwarehouseproject.domain;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import javax.persistence.*;
import java.sql.Date;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Inventory {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
    @OneToOne(fetch = FetchType.EAGER)
    private InventoryState inventoryState;
    @OneToOne(fetch = FetchType.EAGER)
    private Product product;
    @OneToOne(fetch = FetchType.EAGER)
    private Size size;
    private Date soldDate;
    private double soldPrice;
    private double quantity;

}

File: ClothingWarehouseProjectFRONT/src/app/model/inventory.ts
import {InventoryState} from "./inventory-state";
import {Product} from "./product";
import {Size} from "./size";

export class Inventory{

   id: number
   quantity: number
   soldDate: string
   soldPrice: number
   inventoryState: InventoryState
   product: Product
   size: Size


  constructor(){

    this.id = 0
    this.quantity = 0
    this.soldDate = ""
    this.inventoryState = new InventoryState()
    this.soldPrice = 0
    this.product = new Product()
    this.size = new Size()

  }

}
